src/Type/InterfaceType/EnqueuedAsset.php

```diff
@@ -41,38 +41,86 @@ public static function register_type( TypeRegistry $type_registry ) {
 					return ! empty( $type ) ? $type : null;
 				},
 				'fields'      => [
+					'args'         => [
+						'type'              => 'Boolean',
+						'description'       => __( 'Deprecated', 'wp-graphql' ),
+						'deprecationReason' => __( 'Use `EnqueuedAsset.media` instead.', 'wp-graphql' ),
+					],
+					'after'        => [
+						'type'        => [ 'list_of' => 'String' ],
+						'description' => __( 'The inline code to be run after the asset is loaded.', 'wp-graphql' ),
+						'resolve'     => static function ( \_WP_Dependency $asset ) {
+							if ( empty( $asset->extra['after'] ) ) {
+								return null;
+							}
+
+							$after_scripts = array_map(
+								static function ( $after ) {
+									return is_string( $after ) ? $after : null;
+								},
+								$asset->extra['after']
+							);
+
+							return array_filter( $after_scripts );
+						},
+					],
+					'before'       => [
+						'type'        => [ 'list_of' => 'String' ],
+						'description' => __( 'The inline code to be run before the asset is loaded.', 'wp-graphql' ),
+						'resolve'     => static function ( \_WP_Dependency $asset ) {
+							if ( empty( $asset->extra['before'] ) ) {
+								return null;
+							}
+
+							$before_scripts = array_map(
+								static function ( $before ) {
+									return is_string( $before ) ? $before : null;
+								},
+								$asset->extra['before']
+							);
+
+							return array_filter( $before_scripts );
+						},
+					],
+					'conditional'  => [
+						'type'        => 'String',
+						'description' => __( 'The HTML conditional comment for the enqueued asset. E.g. IE 6, lte IE 7, etc', 'wp-graphql' ),
+						'resolve'     => static function ( \_WP_Dependency $asset ) {
+							if ( ! isset( $asset->extra['conditional'] ) || ! is_string( $asset->extra['conditional'] ) ) {
+								return null;
+							}
+
+							return $asset->extra['conditional'];
+						},
+					],
+					'dependencies' => [
+						'type'        => [ 'list_of' => 'EnqueuedScript' ],
```

Comment:
dependencies could be a `Script` or a `Stylesheet`, eh?

```suggestion
						'type'        => [ 'list_of' => 'EnqueuedAsset' ],
```

Reply:
I think you're right 😁,

That said, keep in mind that the `dependency` field wasn't changed (it's in the diff because I sorted the field list).

Also, not sure if this needs to be noted as a breaking change. It looks like all the previously-available `EnqueuedScript` are directly available from the `EnqueuedAsset, and I _think_ a `fragment MyFrag on EnqueuedScript` will work spread on an `EnqueuedAsset` field...

(Leaving this as is so you can review. Maintainer commits are enabled on this PR, so if you're good with the above, feel free to `commit suggestion` and merge no need to wait on me).

Reply:
Ya, this should be considered a bugfix. I'll callout in the release notes a filter should anyone feel like they need the previous behavior.

Reply:
updated the PR description to include this change ✔️

Reply:
Actually, I think we need to make one modification. The Interface says the dependency will be a list_of EnqueuedAsset, but each Type's dependencies field should explicitly return a list_of => EnqueuedScript or list_of => EnqueuedStylesheet. 

Will update.

Reply:
Wont that cause a schema conflict when querying the dependency field for both child types via the interface?

Reply:
@justlevine oh maybe. . . 🤔 

When registering a stylesheet, can it have a JS dependency? And when registering a Script can it have a CSS dependency? 

I feel like I should already know this (and am going to research the answer) but I _thought_ `register_script` dependencies were other scripts and `register_stylesheet` dependencies were always other stylesheets?? 👀

Reply:
![CleanShot 2023-11-14 at 14 18 20](https://github.com/wp-graphql/wp-graphql/assets/1260765/1bbe4391-1d92-4029-805f-dd923a71692d)

Reply:
![CleanShot 2023-11-14 at 14 19 22](https://github.com/wp-graphql/wp-graphql/assets/1260765/5e304a18-267d-46d1-b365-1e38e6e7a415)

Reply:
@justlevine ya, looks like, at least according to docs, the dependencies of a script will always be other scripts, and the dependencies of a stylesheet will always be another stylesheet.

Reply:
Yeah, the deps IRL are always same-type, but because `EnqueuedAsset` is an interface not a union, fields (afaik) with the same name across children need to be defined as the same type and not just another type on a shared interface.

_If so_, then imo the more verbose `...on Enqueued{Script|Stylesheet}`, is probably preferred to a solution that breaks reusable fragments and leads to schema unpredictability by forcing devs to use their own field alias.

Reply:
@justlevine nah, we're good here. 

The Interface `EnqueuedAsset` defines a `dependencies` field that needs to return a `list_of => EnqueuedAsset`

Then the specific `EnqueuedScript.dependencies` returns a `list_of => EnqueuedScript` (which, since EnqueuedScript implements the EnqueuedAsset interface, it passes for properly returning `list_of => EnqueuedAsset`) and same for `EnqueuedStylesheet.dependencies` returning a `list_of => EnqueuedStylesheet`

This is how the `Connection` Interface works. 

Anything that is a `Connection` needs to have a field `nodes` that returns a `list_of => 'Node'` and so any connection that returns `Post` or `Page` or `User` is valid, because all of those things a are a `list_of => Node` too 🙌🏻